docs/src/proposals/fee_transaction_priority.md

```diff
@@ -0,0 +1,212 @@
+---
+title: Fee Transaction Priority
+---
+
+Additional fees were introduced to transactions as a method to allow users to bid for priority for
+their transactions in the leader's queue.
+
+Let the additional fee for a transaction `T` be defined as `F(T)`.
+
+To ensure users get fair priority based on their fee, the proposed scheduler for the leader must
+guarantee that given `T1` and `T2` in the pending queue, and `F(T1) > F(T2)`:
+
+1. `T1` should be considered for processing before `T2`
+2. If `T1` cannot be processed before `T2` because there's already a transaction currently being
+processed that contends on an account `A`, then `T2` should not be scheduled if it would grab
+any account locks needed by `T1`. This prevents lower fee transactions like `T2` from starving
+higher paying transactions like `T1`.
+
+
+### Transaction Pipeline
+
+Pipeline:
+1. Sigverify
+2. Scheduler
```

Comment:
+1 for a separate scheduler.

Reply:
I would add `0. Filter` stage.  once a state auction is saturated, we can start dropping txs that are below the min price to be considered for inclusion.